File: testoniline-services/TSOL.Domain/Entities/Quiz.cs


using System;
using System.Collections.Generic;

namespace TSOL.Domain.Entities
{
    public class Quiz : BaseEntity
    {
        public string Name { get; set; } 
        public string Title { get; set; }
        public string Description { get; set; }
        public float TimeUpMinutes { get; set; }
        public virtual ICollection<Question> Questions { get; set; }
        public DateTime DateCreated { get; set; }
        public DateTime DateModified { get; set; }
        public bool Status { get; set; } = true;
    }
}

File: testonline-ui/src/app/model/quiz.ts
import { QuizConfig } from './quiz-config';
import { Question } from './question';

export class Quiz {

  id: number;
  title: string;
  index: number;
  userId: number ;
  name: string;
  description: string;
  config: QuizConfig;
  date: Date;
  timeUpMinutes: number;
  status: number;
  questions: Question[];

  constructor(data: any) {
    if (data) {
        this.id = data.id;
        this.title = data.title;
        this.index = data.index;
        this.userId = data.userId;
        this.name = data.name;
        this.status = data.status;
        this.timeUpMinutes = data.timeUpMinutes;
        this.description = data.description;
        this.config = new QuizConfig(data.config);
        this.questions = [];
        data.questions.forEach(q => {
            this.questions.push(new Question(q));
        });
    }
  }

}
